src/deriver/deriver.py

```
@@ -185,308 +60,146 @@ async def process_representation_tasks_batch(
+    observations = Representation.from_prompt_representation(
+        response.content,
+        message_ids,
+        latest_message.session_name,
+        latest_message.created_at,
     )
 
-    # Display final observations in a beautiful tree
-    # log_representation(final_observations)
+    if observations.is_empty():
+        logger.warning(
+            f"Deriver generated zero observations for messages {earliest_message.id}:{latest_message.id} in {latest_message.workspace_name}/{latest_message.session_name}!"
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)